### ISSUE TITLE
Various accessibility issues

### ISSUE THREAD

**Issue by alranel** [arduino/arduino-ide#588]
### Describe the problem

This report was shared by Ken Perry on the [developers mailing list](https://groups.google.com/a/arduino.cc/g/developers/c/mf5NDSir1_U/m/q-Y1KdVYAgAJ):

> 1. No way to get to the error window that I can find.
> 2. Open the menus and the menus don't focus.  Note this is with NVDA or Jaws the two most popular screen readers.
> 3. The only thing I can do right now is open code and edit it.  I might be able to compile it and uploaded it with the short cuts but so far I have no idea because I can't get to the error / messages window to tell me what is going on.
> 4. There is no way to get to any other window but the main window. Something like alt0 -alt 9 would help if each one focused different windows.  It might not be the best way to go about things but it would be a short cut start.
> 5. I can change from tab in the edit box and out but when I switch with control +m and then try to tab around the interface nothing happens.
> 6. There is no view window to open and close windows that I need or do not need.  Not that I can get to them anyway.

### Expected behavior

All operations in Arduino IDE can be accomplished using the keyboard alone.

### Arduino IDE version

2.0.0-beta.12

### Operating system

[All](https://groups.google.com/a/arduino.cc/g/developers/c/mf5NDSir1_U/m/yzsN4r_cBwAJ)

### Operating system version

Not stated

### Additional context

We need to investigate more and eventually split them into multiple issues. Some of these may be already tracked in [this repository](https://github.com/arduino/arduino-ide/issues?q=is%3Aopen+is%3Aissue+label%3A%22topic%3A+accessibility%22) or [Theia's](https://github.com/eclipse-theia/theia/issues?q=is%3Aopen+is%3Aissue+label%3Aaccessibility).

**Comment by per1234** [comment]
> 1. No way to get to the error window that I can find.

From reading [the discussion on the Arduino Developers mailing list](https://groups.google.com/a/arduino.cc/g/developers/c/mf5NDSir1_U), I see that "No way" means "no way to accomplish it via the keyboard" and I interpret "error window" as meaning the "**Output**" view. This, and the deficiency as it applies generally to the difficulty in accessing all panels and views is tracked here: https://github.com/arduino/arduino-ide/issues/1561

> 2. Open the menus and the menus don't focus. Note this is with NVDA or Jaws the two most popular screen readers.

I am not able to reproduce this using NVDA, even with the Arduino IDE 2.0.0-beta.12 that was the latest at the time of the report.

> 3. The only thing I can do right now is open code and edit it. I might be able to compile it and uploaded it with the short cuts but so far I have no idea because I can't get to the error / messages window to tell me what is going on.

Tracked at https://github.com/arduino/arduino-ide/issues/1561

> 4. There is no way to get to any other window but the main window. Something like alt0 -alt 9 would help if each one focused different windows. It might not be the best way to go about things but it would be a short cut start.

Tracked at https://github.com/arduino/arduino-ide/issues/1561

> 5. I can change from tab in the edit box and out but when I switch with control +m and then try to tab around the interface nothing happens.

Tracked at https://github.com/arduino/arduino-ide/issues/1561

> 6. There is no view window to open and close windows that I need or do not need. Not that I can get to them anyway.

All views can be opened and closed via the "**Toggle \_\_\_ View**" commands, which can be assigned arbitrary keyboard shortcuts by the user or accessed via the **Command Palette**.

These methods may not be considered a sufficient alternative to a mechanism available via the primary GUI. However, the primary GUI already provides the capability to open and close any view. So the problem is not that the UI doesn't exist, but that the existing UI is not accessible. I think it would be better to focus on making the existing UI elements accessible via keyboard rather than adding additional redundant UI components as a bandaid for the current lack of accessibility.

My impression is that https://github.com/arduino/arduino-ide/issues/360 and https://github.com/arduino/arduino-ide/issues/1561 provide reasonably good coverage of the problems that currently harm the accessibility of those elements.
